lib/spack/spack/environment.py

```diff
@@ -301,27 +302,33 @@ def all_environments():
         yield read(name)
 
 
-def validate(data, filename=None):
-    import jsonschema
-    try:
-        spack.schema.Validator(spack.schema.env.schema).validate(data)
-    except jsonschema.ValidationError as e:
-        raise spack.config.ConfigFormatError(
-            e, data, filename, e.instance.lc.line + 1)
+def _validate_config(data, schema):
+    t = tempfile.NamedTemporaryFile()
```

Comment:
This feels hacky. Could we expose the _mark_overrides function from `config` somehow to handle this inline instead of having to turn everything into a file to be able to read it?

Another option would be to move some of this logic to a method in `spack.config` called `validate_env` or something of the sort and handle that logic there. That might be preferable because it isolates the weirdness of spack config yaml from the environment.